unittests/allplatformstests.py

```diff
@@ -393,6 +394,24 @@ def test_static_compile_order(self):
         self.assertTrue(compdb[3]['file'].endswith("libfile4.c"))
         # FIXME: We don't have access to the linker command
 
+    def test_replace_discouraged_xml_chars(self):
+        '''
+        Test that discouraged xml chars are replaced with their
+        printable representation
+        https://github.com/mesonbuild/meson/pull/10680
+        '''
+        broken_xml_string = ''
+        correct_xml_string = ''
+        for (low, high) in discouraged_unichrs:
+            temp_low = low
+            for uni_chr in range(temp_low, high + 1):
+                char = chr(uni_chr)
+                broken_xml_string = broken_xml_string + char
+                correct_xml_string = correct_xml_string + repr(char)[1:-1]
+        broken_xml_stream = broken_xml_string.encode()
+        decoded_broken_stream = decode(broken_xml_stream)
+        self.assertEqual(decoded_broken_stream, correct_xml_string)
```

Comment:
I'm not sure what's being tested here? That you can code the same logic twice? We should probably use mocked test data and check that it doesn't trigger the bug. For example, by:
- creating a unit test that actually runs meson setup && ninja test && xmllint the-junit-xml
- passing known invalid input into mtest.decode() and checking that it matches what we expect it to be transformed to